From fc2edb6c59a7f11c76c48a1e425e69b1084a9d71 Mon Sep 17 00:00:00 2001
From: Mickey Sola <msola@sourcefire.com>
Date: Fri, 27 Oct 2017 17:04:32 -0400
Subject: bb11945 - fixing null dereference of blob pointer

Patch-Name: bb11945-fixing-null-dereference-of-blob-pointer.patch
---
 libclamav/mbox.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libclamav/mbox.c b/libclamav/mbox.c
index 3df2ae0..9926896 100644
--- a/libclamav/mbox.c
+++ b/libclamav/mbox.c
@@ -2837,6 +2837,10 @@ rfc2047(const char *in)
 				break;
 		}
 		b = messageToBlob(m, 1);
+                if (b == NULL) {
+                    messageDestroy(m);
+                    break;
+                }
 		len = blobGetDataSize(b);
 		cli_dbgmsg("Decoded as '%*.*s'\n", (int)len, (int)len,
 			(const char *)blobGetData(b));
